boards/arduino-leonardo/leonardo-runner.sh

```diff
@@ -1,6 +1,39 @@
 #!/usr/bin/env sh
 set -e
 
+LINUX_OS="LINUX"
+MAC_OS="MAC"
+UNKNOWN_OS="UNKNOWN"
+
+case "$(uname -s)" in
+    Linux*)     OS=$LINUX_OS;;
+    Darwin*)    OS=$MAC_OS;;
+    *)          OS=$UNKNOWN_OS
+esac
```

Comment:
I'd remove the OS variables and just use strings directly:

```suggestion
case "$(uname -s)" in
    Linux*)     OS="Linux";;
    Darwin*)    OS="Mac";;
    *)          OS="Unknown";;
esac
```